### DISCUSSION TITLE
[MSW] Failed to execute `setupWorker` in a non-browser environment.

### DISCUSSION THREAD

**Discussion by dsathyakumar** [mswjs/msw#873] [Q&A]
Hello 👋 
I'm on Jest ^27 and msw latest.
My test env is jsdom. However, upon using setupWorker, I get

```js
[MSW] Failed to execute `setupWorker` in a non-browser environment. . Consider using `setupServer` for Node.js environment instead.
```
I was assuming this would work as it was in a jsdom env. 
Please let me know how this can be resolved.

**Comment by kettanaito** [comment]
Hey, @dsathyakumar.

The error you get is correct: you shouldn't be using `setupWorker` in Jest. Jest is a Node.js process, regardless of `jsdom-environment` (that only controls applied polyfills). 

Please use [`setupServer`](https://mswjs.io/docs/api/setup-server) whenever you want to run MSW in Node.js.
